src/Models/Models.jl

```diff
@@ -25,15 +25,15 @@ import Oceananigans.TimeSteppers: reset!
 function reset!(model::AbstractModel)
 
     for field in fields(model)
-        fill!(field, 0.0)
+        fill!(field, zero(model.grid))
     end
 
     for field in model.timestepper.G⁻
```

Comment:
Though assuming `model.timestepper` is obviously more problematic... (but beyond the scope of this PR I guess)